graphics/palette.h

```diff
@@ -111,9 +111,41 @@ class PaletteManager : Common::NonCopyable {
 
 namespace Graphics {
 
+struct Palette {
+	byte data[256 * 3];
+	uint size;
+
+	/**
+	 * @brief Construct a new Palette object
+	 *
+	 * @param s   the number of palette entries
+	 */
+	Palette(uint s = 0);
+
+	Palette(const Palette &p);
+
+	/**
+	 * @brief Construct a new Palette object
+	 *
+	 * @param p   the palette data, in interleaved RGB format
+	 * @param s   the number of palette entries to be read
+	 */
+	Palette(const byte *p, uint s);
+
+	bool equals(const Palette &p) const;
```

Comment:
Palette should also include grabPalette and setPalette methods to handle subsets of the palette as described in PaletteManager